### PR TITLE
Fix support const as input to linear layers in pytorch

### PR DESCRIPTION
**Pull request by lapid92**
## Pull Request Description:
The issue is that positional weights are only inserted as inputs to the node if the positional weights of the node are being quantized, and in any case where the node weights are not being quantized.
The PR addresses the scenario where the node weights are being quantized and the positional weights are not quantized.

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: tests/pytorch_tests/model_tests/feature_models/const_representation_test.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def __init__(self, unit_test, func, const, input_reverse_order=False):
     def get_tpc(self):
         tp = generate_test_tp_model({'weights_n_bits': 32,
                                      'activation_n_bits': 32,
-                                     'enable_weights_quantization': False,
+                                     'enable_weights_quantization': True,
```
Comment (elad-c) [inline tests/pytorch_tests/model_tests/feature_models/const_representation_test.py:59]: maybe test with AND without weights quantization?

**Comment by ofirgo** [comment]
@lapid92 
Please add description to the PR explaining the problem and the solution (and consider including it in the commit message when merging).

File: model_compression_toolkit/core/common/graph/base_node.py [hunk 2]
```diff
@@ -239,7 +239,9 @@ def insert_positional_weights_to_input_list(self, input_tensors: List) -> List:
         for pos, weight in sorted((pos, weight) for pos, weight in self.weights.items()
                                   if isinstance(pos, int)):
             assert pos <= len(input_tensors), 'Positional weight index mismatch'
-            input_tensors.insert(pos, weight)
+            # Insert only positional weights that are not subject to quantization.
```
Comment (ofirgo) [inline model_compression_toolkit/core/common/graph/base_node.py:242]: maybe explain why?

File: model_compression_toolkit/core/common/graph/base_node.py [hunk 3]
```diff
@@ -239,7 +239,9 @@ def insert_positional_weights_to_input_list(self, input_tensors: List) -> List:
         for pos, weight in sorted((pos, weight) for pos, weight in self.weights.items()
                                   if isinstance(pos, int)):
             assert pos <= len(input_tensors), 'Positional weight index mismatch'
```
Comment (ofirgo) [inline model_compression_toolkit/core/common/graph/base_node.py:241]: change this to if with a "critical" logger message

**Review by ofirgo — CHANGES_REQUESTED** [review]

**Review by ofirgo — APPROVED** [review]

**Pull request merged**